src/eduid_webapp/email/views.py

```diff
@@ -213,29 +216,31 @@ def verify_link(user):
     email = request.args.get('email')
     if code and email:
         current_app.logger.debug('Trying to save email address {} as verified for user {}'.format(email, proofing_user))
-
         db = current_app.proofing_statedb
-        state = db.get_state_by_eppn_and_email(proofing_user.eppn, email)
+        state = db.get_state_by_eppn_and_email(proofing_user.eppn, email, raise_on_missing=False)
+
+        if state is None:
+            current_app.logger.info("Missing state for verification code received for email {} "
+                                    "and user {}.  Sending new code.".format(email, user))
+            return redirect(current_app.config['SAML2_LOGIN_REDIRECT_URL'])
 
         timeout = current_app.config.get('EMAIL_VERIFICATION_TIMEOUT', 24)
         if state.is_expired(timeout):
             current_app.logger.info("Verification code is expired for: {}. Sending new code".format(
```

Comment:
The call to send_verification_code is removed below, so log message would be misleading.

Reply:
fixed